File: test/examples/bool_or_sample_sat/bool_or_sample_sat.cc

























#include <stdlib.h>

#include "absl/types/span.h"
#include "ortools/sat/cp_model.h"

namespace operations_research {
namespace sat {

void BoolOrSampleSat() {
  CpModelBuilder cp_model;

  const BoolVar x = cp_model.NewBoolVar();
  const BoolVar y = cp_model.NewBoolVar();
  cp_model.AddBoolOr({x, ~y});
  

  cp_model.AddBoolOr({x, ~y});
}

}  

}  


int main() {
  operations_research::sat::BoolOrSampleSat();

  return EXIT_SUCCESS;
}

File: test/ok/bool_or_sample_sat.ts
import { operations_research as op } from '../../src'

function test()
{
    let cp_model = new op.sat.CpModelBuilder();
    let x = cp_model.NewBoolVar();
    let y = cp_model.NewBoolVar();
    cp_model.AddBoolOr([x, y.Not()]);
}
test();
